src/backend/parser/cypher_analyze.c

```diff
@@ -612,10 +623,18 @@ static Query *analyze_cypher_and_coerce(List *stmt, RangeTblFunction *rtfunc,
     // check the number of attributes first
     if (attr_count != rtfunc->funccolcount)
     {
-        ereport(ERROR,
-                (errcode(ERRCODE_DATATYPE_MISMATCH),
-                 errmsg("return row and column definition list do not match"),
-                 parser_errposition(pstate, exprLocation(rtfunc->funcexpr))));
+        if (age_guc_enable_auto_column_conversion)
```

Comment:
Should the functionality be: if the column definition list doesn't match: check the flag, or if the flag is enabled don't check the column definition list?

Reply:
Regarding this logic, it's a little uncomfortable while using it, so I'll revise it as follows.
If the corresponding flag is set, it automatically matches regardless of the number of columns, otherwise, if the number of columns does not match, it will cause an error as before.